src/ping.js

```diff
@@ -32,12 +32,19 @@ Ping.prototype.ping = function(source, callback) {
         var pong = new Date() - start;
 
         if (typeof callback === "function") {
-            if (e.type === "error") {
-                console.error("error loading resource");
-                return callback("error", pong);
+
+            if (typeof callback === "function") {
+                if (typeof e === 'undefined' || typeof e.type === 'undefined') {
+                    console.error("timeout error");
+                    return callback("timeout", pong);
+                }
+                if (e.type === "error") {
+                    console.error("error loading resource");
+                    return callback("error", pong);
+                }
+                //return callback(null, pong); // commented to give out a more informative string
+                return callback("success", pong);
```

Comment:
I like the pattern of returning err as null when successful, that way we can use `if (err)` when error checking on this module as shown in the readme example.